bundler/lib/bundler/definition.rb

```diff
@@ -224,7 +224,7 @@ def requested_dependencies
 
     def current_dependencies
       dependencies.select do |d|
-        d.should_include? && !d.gem_platforms(@platforms).empty?
+        d.should_include? && !d.gem_platforms([generic_local_platform]).empty?
```

Comment:
the second conditional is now true 100% of the time, right? because generic local platform will always exist, so empty will always be false, so the condition will always be true?

Reply:
If the current generic local platform is "ruby" and the dependency is for jruby (like `gem "activerecord-jdbcsqlite3-adapter", :platform => :java`), then `d.gem_platforms([generic_local_platform])` will be empty (and the dependency will be filtered out).

Reply:
Although on the other hand, I think that case is already filtered out by `d.should_include?`. Let me run the specs removing that second condition and see what fails (if anything).

Reply:
When removing the second condition, one spec fails (maybe more will fail on Windows/JRuby CI)

https://github.com/rubygems/rubygems/blob/dd23fa18343fd49f65199b1b2b5950edf7f0d16c/bundler/spec/runtime/platform_spec.rb#L364-L379

However, I'm not sure if that may be due to another issue, or whether the expectation may be wrong.

If that's ok, I'll reticket this to not block this PR on that.

Reply:
Yep, totally ok. No need to remove the condition, since my reading on it was wrong anyway!

Reply:
Awesome, I'm merging this now then!